core/tests/protractor_utils/CollectionEditorPage.js

```diff
@@ -63,6 +63,8 @@ var CollectionEditorPage = function() {
   };
 
   // Search and add existing exploration to the node graph.
+  // NOTE TO DEVELOPERS: This assumes that all explorations have unique names
+  // that do not share any words.
```

Comment:
This is not a safe assumption right now. I think the easiest way to make this work would be to add underscores between words for all the explorations we create in the E2E tests

Reply:
I'm leaving a blocking review to make sure we remember to do this

Reply:
@U8NWXD Thanks for pointing this out. I thought about it and I don't think that will work. There is no guarantee that future developers will know to do this, so it's very risky.

Instead, I delved into how to have the search selection be done properly, and I think I've fixed it. PTAL.

Thanks!